packages/ketcher-react/src/script/ui/component/form/input.tsx

```diff
@@ -275,47 +311,62 @@ function ctrlMap(component, { schema, multiple, onChange }) {
   return singleSelectCtrl(component, schema, onChange)
 }
 
-function componentMap({ schema, type, multiple }) {
-  if (schema?.type === 'boolean' && schema?.description === 'slider') {
-    return Slider
-  }
-
+function componentMap(props: Props) {
+  const { schema, type, multiple } = props
   if (!schema || (!schema.enum && !schema.items && !Array.isArray(schema))) {
     if (type === 'checkbox' || (schema && schema.type === 'boolean')) {
       return CheckBox
     }
 
     return type === 'textarea' ? TextArea : GenericInput
   }
+
+  if (schema?.type === 'boolean' && schema?.description === 'slider') {
+    return Slider
+  }
+
   if (multiple || schema.type === 'array')
     return type === 'checkbox' ? FieldSet : Select
 
   return type === 'radio' ? FieldSet : Select
 }
 
-function shallowCompare(a, b) {
-  for (const key in a) {
-    if (!(key in b)) return true
-  }
-  for (const key in b) {
-    if (a[key] !== b[key]) return true
+class Input extends PureComponent<
+  Props & { innerRef: React.Ref<HTMLInputElement> }
+> {
+  component: any
```

Comment:
```suggestion
  component: React.Component
```

Reply:
Unfortunately, it's not that simple =( I'll try once again to set proper type

Reply:
There is an issue with **schema** here - it probably, needs some refactoring - type for component should look smth like this:
ComponentType<{}> | typeof TextArea | typeof GenericInput | typeof Slider | typeof FieldSet | typeof Select
but for some of these ^^ components schema is required, but for class Input itself it is optional (so if we set type like this, we get an error  - for inner component with ref - that schema is missing
Propose to leave it like this for subsequent refactoring